### PR TITLE
feat(avoidance): don't avoid objects around crosswalks

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

- ignore objects around crosswalk in avoidance module. (now, ignore only objects that are stopping for traffic light.)
- https://github.com/autowarefoundation/autoware_launch/pull/362

## Tests performed

rviz_screenshot_2023_05_19-11_46_44
rviz_screenshot_2023_05_19-11_47_20

Psim

## Effects on system behavior

Nothing.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by tkimura4 — APPROVED** [review]
LGTM

File: planning/behavior_path_planner/src/utils/avoidance/utils.cpp [hunk 1]
```diff
@@ -755,18 +755,36 @@ void filterTargetObjects(
...
+
+    if (stop_time_longer_than_threshold && parameters->enable_force_avoidance_for_stopped_vehicle) {
+      // force avoidance for stopped vehicle
+      bool not_parked_object = true;
+
+      // check traffic light
       const auto to_traffic_light =
         utils::getDistanceToNextTrafficLight(object_pose, data.current_lanelets);
+      {
+        not_parked_object = to_traffic_light < parameters->object_ignore_distance_traffic_light;
```
Comment (tkimura4) [inline planning/behavior_path_planner/src/utils/avoidance/utils.cpp:769]: Does not this line need std::abs?

File: planning/behavior_path_planner/src/utils/avoidance/utils.cpp [hunk 2]
```diff
@@ -755,18 +755,36 @@ void filterTargetObjects(
...
 
-      o.to_stop_factor_distance = std::min(to_traffic_light, o.to_stop_factor_distance);
-    }
+      // check crosswalk
+      const auto & ego_pose = planner_data->self_odometry->pose.pose;
+      const auto to_crosswalk =
+        utils::getDistanceToCrosswalk(ego_pose, data.current_lanelets, *rh->getOverallGraphPtr()) -
+        o.longitudinal;
+      {
+        const auto object_to_crosswalk = to_crosswalk - o.longitudinal;
```
Comment (tkimura4) [inline planning/behavior_path_planner/src/utils/avoidance/utils.cpp:778]: This variable is not used.

Comment (satoshi-ota) [reply planning/behavior_path_planner/src/utils/avoidance/utils.cpp]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/3767/commits/1565c628885aead3127af250c6dddb5713ead5e0

**Pull request merged**